modules/social_features/social_event/modules/social_event_an_enroll/src/Plugin/Action/SocialEventAnEnrollSendEmail.php

```diff
@@ -103,31 +116,58 @@ public static function create(ContainerInterface $container, array $configuratio
   /**
    * {@inheritdoc}
    */
-  public function execute($entity = NULL) {
-    $this->entity = $entity;
-
-    if ($this->socialEventAnEnrollManager->isGuest($entity)) {
-      $display_name = $this->socialEventAnEnrollManager->getGuestName($entity, FALSE);
-
-      if (!$display_name) {
-        $display_name = $this->t('Guest');
+  public function executeMultiple(array $objects) {
+    $guests = [];
+    foreach ($objects as $key => $entity) {
+      if ($this->socialEventAnEnrollManager->isGuest($entity)) {
+        $guests[$key] = [
+          'email_address' => $entity->field_email->value,
+          'display_name' => $this->getDisplayName($entity),
+        ];
       }
+    }
 
-      $this->configuration['display_name'] = $display_name;
+    if (!empty($guests)) {
+      // Create some chunks and then make queue items.
+      $chunks = array_chunk($guests, 10);
```

Comment:
Could be nice to make this configurable / or even just a setting in settings.php if we were to find a better sweet spot. But still, it's nais.

Reply:
The parent class: \Drupal\social_user\Plugin\Action\SocialSendEmail has todo's for this.
Currently out of scope.